app-emulation/virtualbox-guest-additions/virtualbox_guest_additions-6.1.26.recipe

```diff
@@ -0,0 +1,109 @@
+SUMMARY="VirtualBox kernel modules and user-space tools for Haiku guests"
+DESCRIPTION="VirtualBox is a free x86 virtualization solution allowing a wide \
+range of x86 operating systems such as Windows, DOS, BSD or Linux to run on a \
+Linux system.
+
+This package contains the VirtualBox kernel modules and user-space \
+tools for Haiku guests."
+HOMEPAGE="https://www.virtualbox.org/"
+COPYRIGHT="2003-2017 Oracle Corporation"
+LICENSE="
+	GNU GPL v2
+	CDDL v1
+	"
+REVISION="1"
+SOURCE_URI="http://download.virtualbox.org/virtualbox/$portVersion/VirtualBox-$portVersion.tar.bz2"
+CHECKSUM_SHA256="0212602eea878d6c9fd7f4a3e0182da3e4505f31d25f5539fb8f7b1fbe366195"
+SOURCE_DIR="VirtualBox-$portVersion"
+PATCHES="virtualbox_guest_additions-$portVersion.patchset"
+
+ARCHITECTURES="!x86_gcc2 ?x86 x86_64"
+SECONDARY_ARCHITECTURES="?x86"
+
+PROVIDES="
+	virtualbox_guest_additions$secondaryArchSuffix = $portVersion
+	cmd:VBoxControl$secondaryArchSuffix = $portVersion
+	cmd:VBoxService$secondaryArchSuffix = $portVersion
+	cmd:VBoxTray$secondaryArchSuffix = $portVersion
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libcurl$secondaryArchSuffix
+	lib:libiconv$secondaryArchSuffix
+	lib:libpng16$secondaryArchSuffix
+	lib:libz$secondaryArchSuffix
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:libcurl$secondaryArchSuffix
+	devel:libiconv$secondaryArchSuffix
+	devel:libpng$secondaryArchSuffix
```

Comment:
Try to use libpng16 here also as mentioned in the issue https://github.com/haikuports/haikuports/issues/3779 : 
```diff
- 	devel:libpng$secondaryArchSuffix
+ 	devel:libpng16$secondaryArchSuffix
```